### DISCUSSION TITLE
How do I Sort Blog Posts ?

### DISCUSSION THREAD

**Discussion by ghost** [imfing/hextra#127] [Q&A]
How can I sort blog posts by weight? 

[This post](https://discourse.gohugo.io/t/how-do-i-sort-by-weight-then-title/18297/4) says that posts are sorted by weight then by linkTitle / Title.

My front matter is:

```
---
title: "Project Name"
type: blog
date: 2023-10-03T22:20:02-07:00
draft: false
toc: true
weight: 1
excludeSearch: false
---
```

and the result looks to be sorted by most recently created instead of weight.

**Comment by imfing** [comment]
Try override the blog list layout under `layouts/blog/list.html`

https://github.com/imfing/hextra/blob/55ff819daeab4eecfcf785d6edfa00e49691e890/layouts/blog/list.html#L9

Copy paste the entire above file under your own site. Change the `range .Pages.ByDate.Reverse` to `range .Pages.ByWeight`

**Comment by ghost** [comment]
Thanks Xin!

The posts are listed by weight. 

But I noticed that when I scroll to the bottom, the sequence of previous and next is not by weight but by create date.

**Reply by imfing** [reply]
In that case, please set `prev` and `next` manually in the frontmatter, for example:

https://github.com/imfing/hextra/blob/55ff819daeab4eecfcf785d6edfa00e49691e890/exampleSite/content/docs/getting-started.md?plain=1#L4-L5

**Reply by ghost** [reply]
What would be the `prev` for the first post and `next` for the last post ?

**Reply by ghost** [reply]
It's not working correctly. `prev` and `next` posts are on the opposite side of where they should be. 

Please try running [this project](https://codeberg.org/mishra/runbook.nu).

**Reply by iju707** [reply]
If you're using blog layout, "reversePagination" will be set to true.
so It is normal for the display to be reversed.
and Empty previous/next are not supported yet.

**Reply by ghost** [reply]
I felt this may be the case - that's why I created issue #129
